CASMHMS-5693: Clear out redfish event subscriptions when add or removing blades (#2324)
Cray-HPE/docs-csm: operations/node_management/Swap_a_Compute_Blade_with_a_Different_System.md

   cray hsm inventory redfishEndpoints update --enabled false x9000c3s0b1
   ```

> ### Source: Clear out the existing Redfish event subscriptions from the BMCs on the blade

Reviewer: ```suggestion
### Source: Clear Redfish event subscriptions from BMCs on the blade
```

> 1. (`ncn#`) Set the environment variable `SLOT` corresponding to the blades location:

Reviewer: ```suggestion
1. (`ncn#`) Set the environment variable `SLOT` to the blade's location.
```

     ```bash

>      ncn# SLOT="x9000c3s0"

Reviewer: ```suggestion
     SLOT="x9000c3s0"
```

     ```

> 1. (`ncn#`) Clear the Redfish event subscriptions:

Reviewer: ```suggestion
1. (`ncn#`) Clear the Redfish event subscriptions.
```

    ```bash

>     ncn# for BMC in $(cray hsm inventory  redfishEndpoints list --type NodeBMC --format json | jq .RedfishEndpoints[].ID -r | grep $SLOT); do

Reviewer: ```suggestion
    for BMC in $(cray hsm inventory  redfishEndpoints list --type NodeBMC --format json | jq .RedfishEndpoints[].ID -r | grep $SLOT); do
```

>         PASSWD=$(cray scsd bmc creds list --targets $BMC --format json | jq .Targets[].Password -r)

Reviewer: ```suggestion
        PASSWD=$(cray scsd bmc creds list --targets ${BMC} --format json | jq .Targets[].Password -r)
```

>         SUBS=$(curl -sk -u root:$PASSWD https://${BMC}/redfish/v1/EventService/Subscriptions | jq -r '.Members[]."@odata.id"')

Reviewer: Would be better to avoid putting the password on the curl command line, because that is visible in the output of `ps -ef` for any users logged into the system, regardless of access level.

>         SUBS=$(curl -sk -u root:$PASSWD https://${BMC}/redfish/v1/EventService/Subscriptions | jq -r '.Members[]."@odata.id"')

Reviewer: ```suggestion
        SUBS=$(curl -sk -u root:"${PASSWD}" https://${BMC}/redfish/v1/EventService/Subscriptions | jq -r '.Members[]."@odata.id"')
```

>         for SUB in $SUBS; do

Reviewer: ```suggestion
        for SUB in ${SUBS}; do
```

            echo "Deleting event subscription: https://${BMC}${SUB}" 

>             curl -i -sk -u root:$PASSWD -X DELETE https://${BMC}${SUB}

Reviewer: Would be better to avoid putting the password on the curl command line, because that is visible in the output of `ps -ef` for any users logged into the system, regardless of access level.

>             curl -i -sk -u root:$PASSWD -X DELETE https://${BMC}${SUB}

Reviewer: ```suggestion
            curl -i -sk -u root:"${PASSWD}" -X DELETE https://${BMC}${SUB}
```

    cray hsm inventory redfishEndpoints update --enabled false x1005c3s0b1
    ```

> ### Destination: Clear out the existing Redfish event subscriptions from the BMCs on the blade

Reviewer: ```suggestion
### Destination: Clear Redfish event subscriptions from BMCs on the blade
```

> 1. Set the environment variable `SLOT` corresponding to the blades location:

Reviewer: ```suggestion
1. (`ncn#`) Set the environment variable `SLOT` to the blade's location.
```

     ```bash

>      ncn# SLOT="x1005c3s0"

Reviewer: ```suggestion
     SLOT="x1005c3s0"
```

     ```

1. Clear the Redfish event subscriptions:

    ```bash

>     ncn# for BMC in $(cray hsm inventory  redfishEndpoints list --type NodeBMC --format json | jq .RedfishEndpoints[].ID -r | grep $SLOT); do

Reviewer: ```suggestion
    for BMC in $(cray hsm inventory  redfishEndpoints list --type NodeBMC --format json | jq .RedfishEndpoints[].ID -r | grep $SLOT); do
```

>         PASSWD=$(cray scsd bmc creds list --targets $BMC --format json | jq .Targets[].Password -r)

Reviewer: ```suggestion
        PASSWD=$(cray scsd bmc creds list --targets ${BMC} --format json | jq .Targets[].Password -r)
```

>         SUBS=$(curl -sk -u root:$PASSWD https://${BMC}/redfish/v1/EventService/Subscriptions | jq -r '.Members[]."@odata.id"')

Reviewer: Would be better to avoid putting the password on the curl command line, because that is visible in the output of `ps -ef` for any users logged into the system, regardless of access level.

>         SUBS=$(curl -sk -u root:$PASSWD https://${BMC}/redfish/v1/EventService/Subscriptions | jq -r '.Members[]."@odata.id"')

Reviewer: ```suggestion
        SUBS=$(curl -sk -u root:"${PASSWD}" https://${BMC}/redfish/v1/EventService/Subscriptions | jq -r '.Members[]."@odata.id"')
```

>         for SUB in $SUBS; do

Reviewer: ```suggestion
        for SUB in ${SUBS}; do
```

            echo "Deleting event subscription: https://${BMC}${SUB}" 

>             curl -i -sk -u root:$PASSWD -X DELETE https://${BMC}${SUB}

Reviewer: ```suggestion
            curl -i -sk -u root:"${PASSWD}" -X DELETE https://${BMC}${SUB}
```
